tokenizer_ts/test/tikTokenizer_p50k_base.test.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as assert from 'assert';
+import * as fs from 'fs';
+import { suite, beforeEach } from 'mocha';
+import {
+    createByEncoderName
+} from '../src/tokenizerBuilder';
+import { TikTokenizer } from '../src/tikTokenizer';
+const IM_START = "<|im_start|>";
+const IM_END = "<|im_end|>";
+const specialTokens: ReadonlyMap<string, number> = new Map([
+    [IM_START, 100264],
+    [IM_END, 100265],
+]);
+
+suite('TikTokenizer p50k_base Test Suite', function () {
+    let tokenizer_p50k_base: TikTokenizer;
+    beforeEach(async () => {
+        tokenizer_p50k_base = await createByEncoderName("p50k_base", specialTokens);
+    });
+
+    test('tokenize source code - p50k_base', () => {
```

Comment:
I'm surprised this works. However we can be more explicit in our wanted behavior.

1. Pass in done() to let Mocha know exactly when the callbacks are done:
```suggestion
    test('tokenize source code - p50k_base', (done) => {
```

```typescript
fs.readfile(filePath, 'utf8', (err, data) => {
    if (err) {
        done(err);
    }
    ...
    done();
```

2. Just use readFileSync
```typescript
const fileContent = fs.readFileSync(filePath, 'utf8')
```